### DISCUSSION TITLE
Could not convert string to float

### DISCUSSION THREAD

**Discussion by Wopples** [JaccoR/hass-entso-e#211] [Q&A]
Here's my template:

```
{% set s = {"B": 13.72,"C": 1.220,"D": 13.72,"E": 0.830,"CF": 0.3333,"CP": 128.99,"ENV": 1.0195,"BHE": 10.89,"WKK": 5.04,"BA": 12.3606,"BTW": 1.21 } %}
{% if states('calendar.belgium') == 'off' %}
  {% if now().isoweekday() < 6 %}
    {% if now().hour >= 7 and now().hour < 22 %}
      {{((s.B+s.C*((1-s.CF)*current_price+s.CF*s.CP))*s.ENV+s.BHE+s.WKK+s.BA)*s.BTW | float }}
    {% endif %}
  {% endif %}
{% else %}
  {{((s.D+s.E*((1-s.CF)*current_price+s.CF*s.CP))*s.ENV+s.BHE+s.WKK+s.BA)*s.BTW | float }}
{% endif %}

```

It works in the developer tools template but when I paste it in in the Entso configuration it returs an error: "could not convert string to float".
What am I missing?
Thanks for any help.

**Comment by rhammen** [comment]
@Wopples For me your template works ok and does not give an error.
I could reproduce the error when I had
{{ current_price }} 
as the current template, and forgot to erase that one, so effectively the template became:
`{{ current_price }} {% set s = {"B": 13.72,"C": 1.220,"D": 13.72,"E": 0.830,"CF": 0.3333,"CP": 128.99,"ENV": 1.0195,"BHE": 10.89,"WKK": 5.04,"BA": 12.3606,"BTW": 1.21 } %}
{% if states('calendar.belgium') == 'off' %}
  {% if now().isoweekday() < 6 %}
    {% if now().hour >= 7 and now().hour < 22 %}
      {{((s.B+s.C*((1-s.CF)*current_price+s.CF*s.CP))*s.ENV+s.BHE+s.WKK+s.BA)*s.BTW | float }}
    {% endif %}
  {% endif %}
{% else %}
  {{((s.D+s.E*((1-s.CF)*current_price+s.CF*s.CP))*s.ENV+s.BHE+s.WKK+s.BA)*s.BTW | float }}
{% endif %}
`
So my hypothesis is that in the copy-pasting of the new template you somehow kept some part of your old template?
